utils/src/semver.rs

```diff
@@ -23,9 +24,14 @@ impl<'de> Deserialize<'de> for Version {
         D: serde::Deserializer<'de>,
     {
         let ver = String::deserialize(deserializer)?;
-        lenient_semver::parse(&ver)
-            .ok()
-            .map(Self)
+        let Ok(mut parsed_ver) = lenient_semver::parse(&ver) else {
+            return Err(D::Error::custom(format!(
+                "Failed to deserialize version {ver}"
+            )));
+        };
+        // delete pre-release field or we can never match pre-release versions of tools
+        parsed_ver.pre = Prerelease::EMPTY;
+        Some(Self(parsed_ver))
             .ok_or_else(|| D::Error::custom(format!("Failed to deserialize version {ver}")))
```

Comment:
```suggestion
        Ok(Self(parsed_ver))
```
Pretty good, just needs to be `Ok` since the return type of the function is `Result`. The `Some` variant is for the `Option` type.

Reply:
I wasn't 100% sure if we wanted a Result with the error intact (e.g. for downstream code to be able to reference it) but I assume it's actually a derived type where the error is just lost because we know the value inside the result isn't None.

Thanks for the review.